storage/innobase/handler/ha_innodb.h

```diff
@@ -499,6 +499,10 @@ class ha_innobase final : public handler
 	this is set in external_lock function */
 	THD*			m_user_thd;
 
+private:
+	/** Row operation counters */
+	struct { size_t deleted= 0, inserted= 0, read= 0, updated= 0; } rows;
```

Comment:
I just noticed it now - the stats are global, not per-table. No need to store this per handler, you can put it in `trx_t`. Then, all the all the tables' stats in the transaction will be updated with a single write. Perhaps, I didnt't consider a performance as an issue here, so didn't notice.

Reply:
Good idea. In that way, we can simply update the (possibly sharded) global statistics in `trx_t::commit()`.